### ISSUE TITLE
💅 `noDuplicateTestHook` produces false positives for Vitest chained `describe.XYZ` blocks

### ISSUE THREAD

**Issue by Bertie690** [biomejs/biome#7205]
### Environment information

```bash
CLI:
  Version:                      2.0.0
  Color support:                true
Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.16.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.12.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExcessiveCognitiveComplexity
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noPrivateImports
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useIsNan
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noConstantBinaryExpression
    nursery/noShadow
    nursery/noTsIgnore
    nursery/useAdjacentGetterSetter
    nursery/useIndexOf
    nursery/useIterableCallbackReturn
    nursery/useObjectSpread
    performance/noAccumulatingSpread
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    performance/noNamespaceImport
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noDescendingSpecificity
    style/noInferrableTypes
    style/noRestrictedTypes
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useBlockStatements
    style/useConst
    style/useDefaultParameterLast
    style/useDeprecatedReason
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useNumberNamespace
    style/useShorthandFunctionType
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAsyncPromiseExecutor
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noOctalEscape
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0
```

### Rule name

noDuplicateTestHook

### Playground link

https://biomejs.dev/playground/?lineWidth=120&indentStyle=space&lintRules=noDuplicateTestHooks&code=ZABlAHMAYwByAGkAYgBlACgAIgBmAG8AbwAiACwAIAAoACkAIAA9AD4AIAB7AAoAIAAgAGIAZQBmAG8AcgBlAEUAYQBjAGgAKAAoACkAIAA9AD4AIAB7AH0AKQA7AAoAIAAgAGIAZQBmAG8AcgBlAEUAYQBjAGgAKAAoACkAIAA9AD4AIAB7AH0AKQA7ACAALwAvACAAZQByAHIAbwByAHMALAAgAGEAcwAgAGUAeABwAGUAYwB0AGUAZAAKACAAIABkAGUAcwBjAHIAaQBiAGUAKAAiAGIAYQByACIALAAgACgAKQAgAD0APgAgAHsACgAgACAAIAAgAGIAZQBmAG8AcgBlAEUAYQBjAGgAKAAoACkAIAA9AD4AIAB7AH0AKQA7ACAALwAvACAAbgBvACAAZQByAHIAbwByACwAIABhAHMAIABlAHgAcABlAGMAdABlAGQACgAgACAAfQApADsACgAKACAAIABkAGUAcwBjAHIAaQBiAGUALgBlAGEAYwBoACgAWwBdACkAKAAiAGIAYQB6ACIALAAgACgAKQAgAD0APgAgAHsACgAgACAAIAAgAGIAZQBmAG8AcgBlAEUAYQBjAGgAKAAoACkAIAA9AD4AIAB7AH0AKQA7ACAALwAvACAAbgBvACAAZQByAHIAbwByACwAIABhAHMAIABlAHgAcABlAGMAdABlAGQACgAgACAAfQApADsACgAKACAAIAAgACAAZABlAHMAYwByAGkAYgBlAC4AZgBvAHIAKABbAF0AKQAoACIAYgBhAHoAIgAsACAAKAApACAAPQA%2BACAAewAKACAAIAAgACAAYgBlAGYAbwByAGUARQBhAGMAaAAoACgAKQAgAD0APgAgAHsAfQApADsAIAAvAC8AIABlAHIAcgBvAHIAPwA%2FAD8APwAKACAAIAB9ACkAOwAKAAoAIAAgAGQAZQBzAGMAcgBpAGIAZQAuAHQAbwBkAG8AKAAiAHEAdQB4ACIALAAgACgAKQAgAD0APgAgAHsACgAgACAAIAAgAGIAZQBmAG8AcgBlAEUAYQBjAGgAKAAoACkAIAA9AD4AIAB7AH0AKQA7ACAALwAvACAAZQByAHIAbwByAD8APwA%2FAAoAIAAgAH0AKQA7AAoACgAgACAAZABlAHMAYwByAGkAYgBlAC4AdABvAGQAbwAuAGUAYQBjAGgAKABbAF0AKQAoACIAYgBhAHoAIgAsACAAKAApACAAPQA%2BACAAewAKACAAIAAgACAAYgBlAGYAbwByAGUARQBhAGMAaAAoACgAKQAgAD0APgAgAHsAfQApADsAIAAvAC8AIABlAHIAcgBvAHIAPwA%2FAD8ACgAgACAAfQApADsACgAKACAAIAAvAC8AIABlAHQAYwAgAGUAdABjACwAIAB0AGgAZQB5ACcAcgBlACAAYQBsAGwAIABsAGkAawBlACAAdABoAGkAcwAKAH0AKQA7AAoA

### Expected result

It should not throw any errors past the first one.

TODO describe blocks are still describe blocks with their own scope for test hooks, as are `describe.for`, `describe.sequential`, etc etc etc.

I tested this with several variants of vitest's `describe.XYZ` and all of them that I tried bork.

Related to #3806 and possibly #3602 as well

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Yeah, seems like a bug to me. I've also noticed that our diagnostic message for this rule could use some improvement too.

**Comment by ToBinio** [comment]
Hi, I was looking into the root cause of this problem and found that 

https://github.com/biomejs/biome/blob/1075b73da739797c0e1a1c3f42c3488017eb973a/crates/biome_js_syntax/src/expr_ext.rs#L1063

Simply does not support `.each` `.todo` `.for` ...

Should this function simple be expanded to detect more of the possible usages?

**Comment by dyc3** [comment]
Yeah, that sounds reasonable. Would you like the issue assigned to you @ToBinio ?

**Comment by ToBinio** [comment]
@dyc3 sure

You noted that the diagnostic message should be updated, what exactly?

**Comment by dyc3** [comment]
It needs to be improved to align more with [our rule pillars](https://biomejs.dev/linter/#rule-pillars).
